pkgs/applications/science/biology/kent/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv
+, libpng
+, libuuid
+, zlib 
+, bzip2
+, lzma 
+, openssl
+, curl
+, libmysqlclient
+, bash
+, fetchFromGitHub
+, which
+}:
+stdenv.mkDerivation rec {
+  pname = "kent";
+  version = "394";
+
+  src = fetchFromGitHub {
+    owner = "ucscGenomeBrowser";
+    repo = pname;
+    rev = "v${version}_base";
+    sha256 = "1ypwrzg5rmzc0b3d3lsx1i63c5cpwwsj0r67mc71263mwh37hgjg";
+  };
+
+  buildInputs = [ libpng libuuid zlib bzip2 lzma openssl curl libmysqlclient ];
+
+  patchPhase = ''
+    substituteInPlace ./src/checkUmask.sh \
+      --replace "/bin/bash" "${bash}/bin/bash"
+    
+    substituteInPlace ./src/hg/sqlEnvTest.sh \
+      --replace "which mysql_config" "${which}/bin/which ${libmysqlclient}/bin/mysql_config"
+  '';
+
+  buildPhase = ''
+    export MACHTYPE=$(uname -m)
+    export CFLAGS="-fPIC"
+    export MYSQLINC=$(mysql_config --include | sed -e 's/^-I//g')
+    export MYSQLLIBS=$(mysql_config --libs)
+    export DESTBINDIR=$NIX_BUILD_TOP/bin
+    export HOME=$NIX_BUILD_TOP
+     
+    cd ./src
+    chmod +x ./checkUmask.sh
+    ./checkUmask.sh
+
+    mkdir -p $NIX_BUILD_TOP/lib
+    mkdir -p $NIX_BUILD_TOP/bin/x86_64
+    
+    make libs
+    cd jkOwnLib
+    make
+
+    cp ../lib/x86_64/jkOwnLib.a $NIX_BUILD_TOP/lib
+    cp ../lib/x86_64/jkweb.a $NIX_BUILD_TOP/lib
+
+    cd ../utils
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+    cp $NIX_BUILD_TOP/lib/jkOwnLib.a $out/lib
+    cp $NIX_BUILD_TOP/lib/jkweb.a $out/lib
+    cp $NIX_BUILD_TOP/bin/x86_64/* $out/bin
+  ''; 
+
+  meta = with stdenv.lib; {
+    description = "UCSC Genome Bioinformatics Group's suite of biological analysis tools, i.e. the kent utilities";
+    license = licenses.mit;
```

Comment:
Based on the discussion it should be `licenses.unfree`. Which is fine to include in nixpkgs, but it will not be built by hydra or be in the cache.

Reply:
Maybe we can encourage upstream to relicense.

Reply:
Thanks for the feedback and sorry about the delay @ryantm,

I managed build only the parts that is mit licensed, which still contains a lot of very useful tools. I will make another PR with these packages only, hopefully at the end of the week.

I can then try to ask for a relicense upstream for the other tools and see what happens :-)

Reply:
@ryantm I decided to go for the unfree license option.

I don't think it matters too much, most bioinf users are used to build from source anyways, and most of them are using it for non-profit I think.

Other libraries using this downstream, most often have similar license mixins as well.